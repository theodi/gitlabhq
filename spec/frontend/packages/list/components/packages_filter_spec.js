import Vuex from 'vuex';
import { GlSearchBoxByClick } from '@gitlab/ui';
import { createLocalVue, shallowMount } from '@vue/test-utils';
import PackagesFilter from '~/packages/list/components/packages_filter.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('packages_filter', () => {
  let wrapper;
  let store;

  const findGlSearchBox = () => wrapper.find(GlSearchBoxByClick);

  const mountComponent = () => {
    store = new Vuex.Store();
    store.dispatch = jest.fn();

    wrapper = shallowMount(PackagesFilter, {
      localVue,
      store,
    });
  };

  beforeEach(mountComponent);

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  it('renders', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('emits events', () => {
    it('sets the filter value in the store on input', () => {
      const searchString = 'foo';
      findGlSearchBox().vm.$emit('input', searchString);

      expect(store.dispatch).toHaveBeenCalledWith('setFilter', searchString);
    });

    it('emits the filter event when search box is submitted', () => {
      findGlSearchBox().vm.$emit('submit');

      expect(wrapper.emitted('filter')).toBeTruthy();
    });
  });
});
