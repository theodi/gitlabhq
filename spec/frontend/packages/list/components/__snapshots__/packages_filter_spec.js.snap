// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`packages_filter renders 1`] = `
<gl-search-box-by-click-stub
  clearable="true"
  clearbuttontitle="Clear"
  clearrecentsearchestext="Clear recent searches"
  closebuttontitle="Close"
  norecentsearchestext="You don't have any recent searches"
  placeholder="Filter by name"
  recentsearchesheader="Recent searches"
  value=""
/>
`;
